Story/977 [IMP] Make RFQs send automatically after scheduler

nocturnalastro opened the pull request.

Commit:
[IMP] udes_purchase: Make RFQs send automatically after scheduler is ran

User-story: story/977
Signed-off:

Commit:
[IMP] udes_purchase: Move RFQ email sender into purchase.order

User-story: story/977
Signed-off:

Commit:
[FIX] udes_purchase: Explicitly commit cursor after running scheduler

Some times the RFQs created on the current run of the check
order points cron job do not show up when searched for within
that cron job.

By explicitly commiting the cursor we ensure that the changes are in
the database.

User-story: story/977
Signed-off:

Commit:
[IMP] udes_purchase: Add custom email and report template for reorder point

We don't require most of the infomation in the standerd attachment.
Also the correct layout was not being used.

Added new email and report, also added context to send_mail call.
Functions for both the reference and context were added to allow
them to be overridden if changes are required.

User-story: story/977

Commit:
[IMP] Remove commented out portions of email template

User-story: 977

ws-mps commented on models/purchase.py:
```diff
@@ -0,0 +1,71 @@
+        Mail = self.env['mail.mail']
+        RequestOrder = self.env['purchase.order']
+
+        email_template = self.get_email_template()
```
Looks like we should be calling: get_rfq_email_template

Commit:
[FIX] Change name of function call to match function name

User-story: 977

Commit:
[IMP] udes_purchase: Use a copy of context in email render

Without copy the context could be modified which would be hard to debug

User-story: 977
